uni/ulsp/lsif.icn

```diff
@@ -0,0 +1,392 @@
+   #  to a directed file in LSIF JSON format.
+   method build_index()
+      local i, j, k, idoc, path, x, pack, file_in_pack, uri, filehandler_object, fake_params, result, hoverHandler,
+            context, char_end, line_end, definitionHandler, result2, resultSet_table, resultSet, reference,
+            reference_table, c, b := 0, test := 0
+      hoverHandler := HoverHandler()
+      definitionHandler := DefinitionHandler()
+      resultSet_table := table()
+      reference_table := table()
+      fake_params := table()
+      fake_params["position"] := table()
+      createMetaDataVertex(db.unicon_dir)
+      every i := 1 to *db.dir_list do {
+         path := db.dir_list[i]
+
+         idoc := UniAll()
+
+         idoc.setSourcePath(path)
+         idoc.setTargetDir(path)
+         idoc.processFile(path)
+         x := idoc.packages
+         path := replace(path, "C:", "c:")
+         path ? {
+            &pos := 0
+            while c := move(-1) do {
+               if c == "/" | c == "\\" then b +:= 1
+               if b == 2 then break
+            }
+            createProjectVertex(path, path[&pos + 1:0])
+         }
+         createEventVertex("project", "begin", projectID)
+         document_list := []
+         every pack := !x do {
+            every file_in_pack := pack.getFiles().get() do {
+               fake_params["position"]["line"] := 0
+               fake_params["position"]["character"] := 0
+               range_list := []
+               uri := path || "/" || file_in_pack.getName()
+
+               Logger.info("[LSIF] Processing: " || uri)
+               createDocumentVertex(uri)
+               createEventVertex("document", "begin", documentID)
+               filehandler_object := FileHandler(uri, db)
+               filehandler_object.setFile()
+               filehandler_object.updateFileHandler()
+               j := 0
+               k := 0
+               context := filehandler_object.getContext(1, 1)
+               line_end := context.totalFileLines
+               while j < line_end do {
+                  fake_params["position"]["line"] := j
+                  context := filehandler_object.getContext(j+1, 1)
+                  char_end := *context.line
+                  while k < char_end do {
+                     fake_params["position"]["character"] := k
+                     hoverHandler.setVariables(filehandler_object, fake_params)
+                     definitionHandler.setVariables(filehandler_object, fake_params)
+                     result := hoverHandler.run("lsif")
+                     if \result & tojson(result) ~== "{}" then {
+                        createRangeVertex(result["line"], result["begins"], result["ends"])
+                        result2 := definitionHandler.run()
+                        if \result2 & tojson(result2) ~== "{}" & result2["uri"] == uri then {
+                           if (j >= result2["range"]["start"]["line"] && j <= result2["range"]["end"]["line"] &&
```

Comment:
In unicon, you can write this in a more natural way (`j` is between those numbers) as:
```
if ( result2["range"]["start"]["line"] <=  j <= result2["range"]["end"]["line"]
```